### PR TITLE
Add timeout wait parameter (mlagents_envs)

### PR DESCRIPTION
**Pull request by awjuliani**
* Adds timeout wait parameter to control how long python interface waits for Unity environment to connect.

### CODE REVIEW

File: ml-agents-envs/mlagents_envs/environment.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ class UnityEnvironment(object):
 
     def __init__(self, file_name=None, worker_id=0,
                  base_port=5005, seed=0,
-                 docker_training=False, no_graphics=False):
+                 docker_training=False, no_graphics=False,
+                 timeout_wait=30):
```
Comment (harperj) [inline ml-agents-envs/mlagents_envs/environment.py]: personally I'd call this something like `communicator_timeout` or `connection_timeout` -- no big deal if you think this is easier to understand though.

File: ml-agents-envs/mlagents_envs/environment.py [hunk 2]
```diff
@@ -207,8 +209,8 @@ def executable_launcher(self, file_name, docker_training, no_graphics):
                                               stderr=subprocess.PIPE,
                                               shell=True)
 
-    def get_communicator(self, worker_id, base_port):
-        return RpcCommunicator(worker_id, base_port)
+    def get_communicator(self, worker_id, base_port, timeout_wait):
```
[the remark this answers is not in the document]
Comment (harperj) [reply ml-agents-envs/mlagents_envs/environment.py]: I'd go even further and say we might as well just remove this method and use `self.communicator = RpcCommunicator(...)` if we're not supporting `SocketCommunicator` (which I think we already have not been).

[the remark this answers is not in the document]
Comment (awjuliani) [reply ml-agents-envs/mlagents_envs/environment.py]: Agree. Just fixed it.

**Comment by awjuliani** [comment]
@harperj Removed the un-necessary function call.

**Pull request merged**